packages/hierarchies-react/src/presentation-hierarchies-react/UseSelectionHandler.ts

```diff
@@ -0,0 +1,165 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useEffect, useRef } from "react";
+import { isPresentationHierarchyNode, PresentationTreeNode } from "./Types";
+
+/** @internal */
+export enum SelectionModeFlags {
+  Single = 1 << 0,
+  DeselectEnabled = 1 << 1,
+  KeysEnabled = 1 << 2,
+  None = 1 << 3,
+}
+
+/** @beta */
+export enum SelectionMode {
+  /** Only one item selected at a time. */
+  Single = SelectionModeFlags.Single,
+  /** Only one item selected at a time; allows deselecting. */
+  SingleAllowDeselect = SelectionModeFlags.Single | SelectionModeFlags.DeselectEnabled,
+  /** Toggles items. */
+  Multiple = SelectionModeFlags.DeselectEnabled,
+  /** Toggles items; allows the use of Ctrl & Shift Keys. */
+  Extended = SelectionModeFlags.KeysEnabled | SelectionModeFlags.DeselectEnabled,
+  /** Disables selection */
+  None = SelectionModeFlags.None,
+}
```

Comment:
@grigasp Do you think we need all of this? Or can we simplify this into `export type SelectionMode = "none" | "single" | "extended" | "multiple".

Reply:
Switched to string union.